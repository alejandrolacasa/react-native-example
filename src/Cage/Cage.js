import * as React from 'react';
import { NavigationScreenProp, NavigationState } from 'react-navigation';
import { Button } from 'react-native';
import Face from '../Face/Face';

import cageSrc from './cage.jpg';

type Props = {
  navigation: NavigationScreenProp<NavigationState, *>,
};

const Cage = ({ navigation }: Props) => (
  <Face
    button={
      <Button
        title="Turn it Travolta"
        onPress={() => navigation.navigate('Travolta')}
      />
    }
    faceSrc={cageSrc}
  />
);

Cage.navigationOptions = {
  title: 'Cage',
  headerStyle: {
    backgroundColor: 'blue',
  },
};

export default Cage;
