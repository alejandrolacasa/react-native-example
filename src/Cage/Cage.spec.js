import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cage from './Cage';

describe('Cage component', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<Cage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call navigation on press', () => {
    const navigationMock = {
      navigate: jest.fn(),
    };

    const wrapper = shallow(<Cage navigation={navigationMock} />);
    const button = shallow(wrapper.props().button);

    button.simulate('press');

    expect(navigationMock.navigate).toHaveBeenCalledWith('Travolta');
  });
});
