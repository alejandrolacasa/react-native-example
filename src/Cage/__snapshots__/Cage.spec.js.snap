// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cage component should render without crashing 1`] = `
<Face
  button={
    <Button
      onPress={[Function]}
      title="Turn it Travolta"
    />
  }
  faceSrc={
    Object {
      "testUri": "../../../src/Cage/cage.jpg",
    }
  }
/>
`;
